OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs

```diff
@@ -21,7 +21,19 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Attach this to a unit to enable dynamic upgrades by warheads, experience, crates, support powers, etc.")]
 	public class UpgradeManagerInfo : ITraitInfo, Requires<IUpgradableInfo>
 	{
-		public object Create(ActorInitializer init) { return new UpgradeManager(init); }
+		[UpgradeGrantedReference]
+		[Desc("Upgrades that can be granted/revoked from Lua scripts.")]
+		public readonly HashSet<string> Scriptable = new HashSet<string>();
+
+		[UpgradeUsedReference]
+		[Desc("Upgrades not to check whether used.")]
+		public readonly HashSet<string> IgnoredAsUnused = new HashSet<string>();
```

Comment:
Maybe a kind of hack, since it is for sticky situations like `eliteweapon` from `GainsExperience` where granted upgrades are inherited but not used and the only other ways to handle it are 1) redefine an entire MiniYaml entry or 2) redefine how `HashSet<string>` can be stored in MiniYaml to allow for removing elements.